packages/pigeon/lib/generator_tools.dart

```diff
@@ -588,6 +619,88 @@ String? deducePackageName(String mainDartFile) {
   }
 }
 
+/// Recursively search for all the interfaces apis from a list of names of
+/// interfaces.
+///
+/// This method assumes that all interfaces names can be found in
+/// [allProxyApis]. Otherwise, throws an [ArgumentError].
+Set<AstProxyApi> recursiveFindAllInterfacesApis(
```

Comment:
Since a user has to write Dart code to do this, I assumed that `pigeon` would throw an error in this situation since it's invalid Dart. However, it looks like `pigeon` still runs in this scenario currently, so I added code to prevent this and added a test.